contracts/src/CrossChainApplications/examples/NativeTokenBridge/NativeTokenDestination.sol

```diff
@@ -215,25 +287,43 @@ contract NativeTokenDestination is TeleporterOwnerUpgradeable, INativeTokenDesti
         require(recipient != address(0), "NativeTokenDestination: zero recipient address");
         require(amount != 0, "NativeTokenDestination: zero transfer value");
 
+        uint256 scaledAmount = _scaleTokens(amount, true);
+
         // If the contract has not yet been collateralized, we will deduct as many tokens
         // as needed from the transfer as needed. If there are any excess tokens, they will
         // be minted and sent to the recipient.
-        uint256 adjustedAmount = amount;
+        uint256 adjustedAmount = scaledAmount;
         if (currentReserveImbalance > 0) {
-            if (amount > currentReserveImbalance) {
+            if (scaledAmount > currentReserveImbalance) {
                 emit CollateralAdded({amount: currentReserveImbalance, remaining: 0});
-                adjustedAmount = amount - currentReserveImbalance;
+                adjustedAmount = scaledAmount - currentReserveImbalance;
                 currentReserveImbalance = 0;
             } else {
-                currentReserveImbalance -= amount;
-                emit CollateralAdded({amount: amount, remaining: currentReserveImbalance});
+                currentReserveImbalance -= scaledAmount;
+                emit CollateralAdded({amount: scaledAmount, remaining: currentReserveImbalance});
                 return;
             }
         }
 
         totalMinted += adjustedAmount;
         emit NativeTokensMinted(recipient, adjustedAmount);
```

Comment:
Think these two lines should be moved under `if adjustedAmount > 0` check as well

Reply:
Michael had suggested emitting an event even in the 0 case, to make it more clear what is happening to the end user.